core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/hashJoin/HashJoinOpDesc.scala

```diff
@@ -37,6 +37,7 @@ class HashJoinOpDesc[K] extends OperatorDescriptor {
 
   override def operatorExecutor: OpExecConfig = {
     opExecConfig = new HashJoinOpExecConfig[K](
+      this.getCachedOutputSchema,
```

Comment:
what if cachedOutputSchema wasn't set for some reason - eg: outputSchema.isDefined in WorkflowCompiler returns false? cachedOutputSchema will be null. Should we do an assert or exception throw somewhere?

Reply:
That's a good point. I think we should do the assertions in `WorkflowCompiler`, so that we're guaranteed to have proper `Schema` objects everywhere.